contracts/Zauction.sol

```diff
@@ -46,22 +48,13 @@ contract ZAuction is Initializable, OwnableUpgradeable {
 
   event BidCancelled(uint256 bidNonce, address indexed bidder);
 
-  function getTopLevelId(uint256 tokenId) private returns (uint256) {
-    uint256 topLevelId = topLevelDomainIdCache[tokenId];
-    if (topLevelId == 0) {
-      topLevelId = topLevelDomainIdOf(tokenId);
-      topLevelDomainIdCache[tokenId] = topLevelId;
-    }
-    return topLevelId;
-  }
-
-  function initialize(IERC20 tokenAddress, IRegistrar registrarAddress)
+  function initialize(IERC20 tokenAddress, IZNSHub znsHubAddress)
     public
     initializer
   {
     __Ownable_init();
     token = tokenAddress;
-    registrar = registrarAddress;
+    hub = znsHubAddress;
```

Comment:
`initialize` can only be called once. Since this contract is already deployed in mainnet, we need to upgrade and then call a different method to set the hub.